drivers/entropy/entropy_sy1xx_trng.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) 2025 sensry.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sensry_sy1xx_trng
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sy1xx_entropy, CONFIG_ENTROPY_LOG_LEVEL);
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/entropy.h>
+#include <errno.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+
+#define SY1XX_TRNG_VAL_OFFS        0x00
+#define SY1XX_TRNG_FIFO_COUNT_OFFS 0x04
+#define SY1XX_TRNG_STATUS_OFFS     0x08
+#define SY1XX_TRNG_ERROR_OFFS      0x0c
+
+#define SY1XX_TRNG_FIFO_SIZE 64
+
+/* time needed to fill fifo when empty */
+#define SY1XX_TRNG_FIFO_REFILL_TIME_USEC   80
+#define SY1XX_TRNG_FIFO_REFILL_MAX_RETRIES 5
+
+struct sy1xx_trng_config {
+	uint32_t base_addr;
+};
+
+struct sy1xx_trng_data {
+	struct k_mutex mutex;
+};
+
+static int sy1xx_trng_driver_init(const struct device *dev)
+{
+	const struct sy1xx_trng_config *const cfg = dev->config;
+	struct sy1xx_trng_data *const data = dev->data;
+
+	k_mutex_init(&data->mutex);
+
+	/* trng comes up fully initialized, so only check if all is fine */
+	if (0 != sys_read32(cfg->base_addr + SY1XX_TRNG_ERROR_OFFS)) {
+		LOG_ERR("failure mode active, internal init failed");
+		return -EINVAL;
+	}
+
+	if (SY1XX_TRNG_FIFO_SIZE != sys_read32(cfg->base_addr + SY1XX_TRNG_FIFO_COUNT_OFFS)) {
+		LOG_ERR("fifo not fully loaded");
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int sy1xx_trng_driver_get_entropy(const struct device *dev, uint8_t *buffer, uint16_t length)
+{
+	const struct sy1xx_trng_config *const cfg = dev->config;
+	struct sy1xx_trng_data *const data = dev->data;
+	uint32_t max_retries = SY1XX_TRNG_FIFO_REFILL_MAX_RETRIES;
+	uint32_t random_word;
+	int ret = 0;
+
+	uint32_t word_count = DIV_ROUND_UP(length, 4);
+
+	for (uint32_t i = 0; i < word_count; i++) {
+
+		while (true) {
+
+			k_mutex_lock(&data->mutex, K_FOREVER);
+
+			/* make sure fifo has at least one random word */
+			if (sys_read32(cfg->base_addr + SY1XX_TRNG_FIFO_COUNT_OFFS) > 0) {
+				/* get new random word */
+				random_word = sys_read32(cfg->base_addr + SY1XX_TRNG_VAL_OFFS);
+				k_mutex_unlock(&data->mutex);
+				break;
+			}
+
+			/* currently no random values available, thus wait */
+			max_retries--;
+			if (max_retries == 0) {
+				k_mutex_unlock(&data->mutex);
+				return -EAGAIN;
+			}
+			k_sleep(K_USEC(SY1XX_TRNG_FIFO_REFILL_TIME_USEC));
+		};
+
+		memcpy(&buffer[i * 4], &random_word, MIN(length, 4));
+		length -= 4;
+	}
+
+	/* always error check, to make sure that we received valid readings */
+	if (0 != sys_read32(cfg->base_addr + SY1XX_TRNG_ERROR_OFFS)) {
+		LOG_ERR("failure mode active, reading of values failed");
+		return -EINVAL;
+	}
+
+	return ret;
+}
+
+static int sy1xx_trng_driver_get_entropy_within_isr(const struct device *dev, uint8_t *buffer,
```

Comment:
Could just be called
```suggestion
static int sy1xx_trng_driver_get_entropy_isr(const struct device *dev, uint8_t *buffer,
```
as the entropy driver API does.